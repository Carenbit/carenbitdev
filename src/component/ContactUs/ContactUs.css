.success {
  position: fixed;
  top: 10vh;
  font-size: 18px;
  color: #fff;
  padding: 10px;
  text-align: center;
  background-color: #fb6738;
  border-radius: 10px;
}

.hide {
  display: none;
}
.carenbit {
  height: 40px;
}

p {
  font-size: 20px;
  font-weight: 400;
  line-height: 27px;
  letter-spacing: -0.02em;
  text-align: left;
}

.footer {
  background-color: #0a0a0a;
  color: #fff;
  padding: 20px;
  text-align: center;
  flex-wrap: wrap;
  height: 80px;
  justify-content: space-between;
}

.mediaIcon {
  font-size: 20px;
  color: #fb6738;
}

.contactBlock {
  display: flex;
  gap: 50px;
  justify-content: space-evenly;
  align-items: center;
  padding: 50px 30px;
  flex-wrap: wrap;
}

.contactForm {
  display: flex;
  flex-direction: column;
  gap: 20px;
  background-color: #fff4f0;
  padding: 50px 30px;
  border-radius: 20px;
}

.contactHeading {
  font-size: 35px;
  font-weight: 600;
  line-height: 48px;
  letter-spacing: -0.02em;
}

.contactBox {
  display: flex;
  flex-direction: column;
  gap: 20px;
  align-items: center;
}

input {
  padding: 10px;
  border: 2px solid #fb6738b2;
}

textarea {
  padding: 10px;
  border: 2px solid #fb6738b2;
}

.contactIcon {
  font-size: 30px;
  color: #fb6738;
}

.spacing {
  padding-left: 20px;
}
@media only screen and (max-device-width: 464px) {
  .contactBlock {
    gap: 20px;
    margin: auto;
  }
  .contactForm {
    gap: 10px;
  }
  .contactIcon {
    font-size: 20px;
  }

  p {
    font-size: 16px;
  }
  .spacing {
    padding-left: 0px;
  }
  .footer {
    height: fit-content;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }
  .mediaIcon {
    font-size: 16px;
  }
}

@media only screen and (min-device-width: 464px) and (max-device-width: 768px) {
  .contactBlock {
    gap: 20px;
    flex-direction: column;
  }
  .contactForm {
    gap: 10px;
    min-width: 400px;
  }
  .footer {
    height: fit-content;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }
}

@media only screen and (min-device-width: 768px) {
  .contactForm {
    gap: 10px;
    min-width: 400px;
  }
}

@media only screen and (min-device-width: 1024px) {
  .contactForm {
    gap: 20px;
    min-width: 500px;
  }
}
