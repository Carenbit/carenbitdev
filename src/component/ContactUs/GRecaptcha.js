// site key:6Ld0tokpAAAAAEHVaDoLyEcCrXPOqUj7vUCfTF9x
// secret key: 6Ld0tokpAAAAADr1HCmB3KywtH8uIHanREpLk4vK

// import React from "react";
// import {
//   GoogleReCaptchaProvider,
//   GoogleReCaptcha,
// } from "react-google-recaptcha-v3";

// const GRecaptcha = (handleToken) => {
//   const [token, setToken] = React.useState(" ");

//   const handleTokenData = (token) => {
//     handleToken(token);
//   };

//   return (
//     <GoogleReCaptchaProvider reCaptchaKey="6LefZ4cpAAAAAMQO_iWeCfmeiTYa49IFLT6W949v">
//       <GoogleReCaptcha
//         onVerify={(token) => {
//           setToken(token);
//           handleTokenData(token);
//         }}
//       />
//     </GoogleReCaptchaProvider>
//   );
// };

// export default GRecaptcha;
