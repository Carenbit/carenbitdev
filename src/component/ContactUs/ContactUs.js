import React from "react";
import "./ContactUs.css";
import { useGoogleReCaptcha } from "react-google-recaptcha-v3";
import {} from "react-google-recaptcha-v3";
import { FaLocationDot, FaXTwitter } from "react-icons/fa6";
import {
  FaPhoneAlt,
  FaLinkedin,
  FaRegCopyright,
  FaWhatsapp,
} from "react-icons/fa";
import { MdEmail } from "react-icons/md";
import { useForm, ValidationError } from "@formspree/react";
import { motion } from "framer-motion";
import carenbit from "../../assets/carenbit.png";

const ContactUs = () => {
  // const [state, handleSubmit] = useForm("xzbnrgyp");

  const [success, setSuccess] = React.useState(false);
  const [error, setError] = React.useState(false);
  const { executeRecaptcha } = useGoogleReCaptcha();
  const [state, handleSubmit] = useForm("xdojgvqp", {
    data: { "g-recaptcha-response": executeRecaptcha },
  });

  const handleFormSubmit = (event) => {
    event.preventDefault();
    console.log(state.errors);
    if (
      !event.target.name.value ||
      !event.target.email.value ||
      !event.target.message.value
    ) {
      setError(true);
      setTimeout(() => {
        setError(false);
      }, 3000);
      return;
    }
    handleReCaptchaVerify(event);
  };

  const handleReCaptchaVerify = React.useCallback(
    (e) => {
      e.preventDefault();
      if (!executeRecaptcha) {
        console.log("Execute recaptcha not yet available");
        return;
      }
      console.log("execute recaptcha is", executeRecaptcha);
      executeRecaptcha("enquiryFormSubmit").then((gReCaptchaToken) => {
        console.log(gReCaptchaToken, "response Google reCaptcha server");
        console.log(e);
      });
      handleSubmit(e);
    },
    [executeRecaptcha, handleSubmit]
  );

  React.useEffect(() => {
    if (state.succeeded) {
      setSuccess(true);
      const formElement = document.getElementById("contactForm");
      formElement.reset();
      setTimeout(() => {
        setSuccess(false);
      }, 5000);
    }
  }, [state.succeeded]);

  return (
    <div id="contactUsPage">
      {success && (
        <>
          <motion.div
            initial={{ opacity: 0, x: "30vw", scale: 0.3 }}
            animate={{ opacity: 1, x: 0, scale: 1 }}
            exit={{ opacity: 0, x: "-30vw", scale: 0.3 }}
            transition={{
              type: "spring",
              stiffness: 120,
            }}
            className="success"
          >
            <p>Thanks for reaching out, will get back to you</p>
          </motion.div>
        </>
      )}
      {error && (
        <>
          <motion.div
            initial={{ opacity: 0, x: "30vw", scale: 0.3 }}
            animate={{ opacity: 1, x: 0, scale: 1 }}
            exit={{ opacity: 0, x: "-30vw", scale: 0.3 }}
            transition={{
              type: "spring",
              stiffness: 120,
            }}
            className="success"
          >
            <p>Please fill all the fields</p>
          </motion.div>
        </>
      )}

      <h1 className="text-center contactHeading">Contact Us</h1>
      <div className="contactBlock">
        <form
          onSubmit={handleFormSubmit}
          className="contactForm"
          id="contactForm"
        >
          <label htmlFor="name" className="form-label">
            Name
          </label>
          <input
            id="name"
            type="name"
            name="name"
            autoComplete="Given name"
            placeholder="Enter Full Name"
            minLength={3}
            maxLength={50}
          />
          <ValidationError
            prefix="Name"
            field="name"
            errors={state.errors}
            className="errorMessage"
          />
          <label htmlFor="email" className="form-label">
            Email
          </label>
          <input
            id="email"
            type="email"
            name="email"
            autoComplete="email"
            placeholder="Enter Email Address"
          />
          <ValidationError
            prefix="Email"
            field="email"
            errors={state.errors}
            className="errorMessage"
          />
          <label htmlFor="message" className="form-label">
            Message
          </label>
          <textarea
            id="message"
            name="message"
            placeholder="Write your Message"
            required
            minLength={10}
          />
          <ValidationError
            prefix="Message"
            field="message"
            errors={state.errors}
            className="errorMessage"
          />
          <button
            type="submit"
            disabled={state.submitting}
            className="mx-auto contactButton"
          >
            Send message
          </button>

          <ValidationError errors={state.errors} className="errorMessage" />
        </form>
        <div className="contactBox">
          <div className="addressBox">
            <a
              href="https://maps.app.goo.gl/N5JFFQ46on3axwAz8"
              target="_blank"
              rel="noreferrer"
            >
              <FaLocationDot className="contactIcon location" />
            </a>
            <div className="address">
              <span>Dhruv Darshan Co-op. Soc</span>
              <span> Sector.No-26 </span>
              <span>Nigdi Pradhikaran </span>Ravet, Tal - Haveli
              <span> PuneÂ 411044</span>
            </div>
          </div>
          <p>
            <a href="tel:917972379031">
              <FaPhoneAlt className="contactIcon" />
              <span className="spacing">+91 79723 79031 </span>
            </a>
          </p>
          <p>
            <a
              href="mailto:carenbit.tech@gmail.com"
              target="_blank"
              rel="noreferrer"
            >
              <MdEmail className="contactIcon" />
              <span className="spacing">carenbit.tech@gmail.com</span>
            </a>
          </p>
        </div>
      </div>
      <div className="footer d-flex ">
        <div>
          <img src={carenbit} alt="carenbit" className="carenbit" />
        </div>
        <p className="mediaIcon">
          <FaRegCopyright />
          2024{" "}
          <strong>
            <em>Carenbit</em>
          </strong>
        </p>
        <div className=" d-flex justify-content-between gap-3">
          <span className="contactDetails">
            <a
              href="https://api.whatsapp.com/send?phone=917972379031"
              target="_blank"
              rel="noreferrer"
            >
              <FaWhatsapp className="mediaIcon zoom" />
            </a>
          </span>
          <span className="contactDetails">
            <a
              href="mailto:carenbit.tech@gmail.com"
              target="_blank"
              rel="noreferrer"
            >
              <MdEmail className="mediaIcon zoom" />
            </a>
          </span>
          <span className="contactDetails">
            <a
              href="https://www.linkedin.com/company/carenbit/"
              target="_blank"
              rel="noreferrer"
            >
              <FaLinkedin className="mediaIcon zoom" />
            </a>
          </span>
          <span className="contactDetails">
            <a
              href="https://twitter.com/carenbit"
              target="_blank"
              rel="noreferrer"
            >
              <FaXTwitter className="mediaIcon zoom" />
            </a>
          </span>
        </div>
      </div>
    </div>
  );
};

export default ContactUs;
