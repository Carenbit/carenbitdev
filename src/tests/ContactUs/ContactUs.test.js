import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import ContactUs from "../../component/ContactUs/ContactUs.js";

test("should have a quick enquiry button", () => {
  render(<ContactUs />);
  const button = screen.getByText(/Send message/i);
  expect(button).toBeInTheDocument();
  expect(
    screen.getByRole("button", { name: /Send message/i })
  ).toBeInTheDocument();
});

test('should have a input field with placeholdeer "Enter Full Name"', () => {
  render(<ContactUs />);
  const input = screen.getByPlaceholderText(/Enter Full Name/i);
  expect(input).toBeInTheDocument();
});

test('should have a input field with placeholdeer "Enter Email Address"', () => {
  render(<ContactUs />);
  const input = screen.getByPlaceholderText(/Enter Email Address/i);
  expect(input).toBeInTheDocument();
});

test('should have a input field with placeholdeer "Write your Message"', () => {
  render(<ContactUs />);
  const input = screen.getByPlaceholderText(/Write your Message/i);
  expect(input).toBeInTheDocument();
});

test("should match screenshot of ContactUS component", () => {
  const { container } = render(<ContactUs />);
  expect(container).toMatchSnapshot();
});
