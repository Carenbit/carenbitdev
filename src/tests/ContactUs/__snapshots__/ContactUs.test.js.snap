// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match screenshot of ContactUS component 1`] = `
<div>
  <div
    id="contactUsPage"
  >
    <h1
      class="text-center contactHeading"
    >
      Contact Us
    </h1>
    <div
      class="contactBlock"
    >
      <form
        class="contactForm"
      >
        <label
          class="form-label"
          for="name"
        >
          Name
        </label>
        <input
          id="name"
          name="name"
          placeholder="Enter Full Name"
          type="name"
        />
        <label
          class="form-label"
          for="email"
        >
          Email
        </label>
        <input
          id="email"
          name="email"
          placeholder="Enter Email Address"
          type="email"
        />
        <label
          class="form-label"
          for="message"
        >
          Message
        </label>
        <textarea
          id="message"
          name="message"
          placeholder="Write your Message"
        />
        <button
          class="mx-auto contactButton"
          type="submit"
        >
          Send message
        </button>
      </form>
      <div
        class="contactBox"
      >
        <p>
          <svg
            class="contactIcon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 384 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M215.7 499.2C267 435 384 279.4 384 192C384 86 298 0 192 0S0 86 0 192c0 87.4 117 243 168.3 307.2c12.3 15.3 35.1 15.3 47.4 0zM192 128a64 64 0 1 1 0 128 64 64 0 1 1 0-128z"
            />
          </svg>
          <span
            class="spacing"
          >
            Pune, Maharashtra
          </span>
        </p>
        <p>
          <svg
            class="contactIcon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M497.39 361.8l-112-48a24 24 0 0 0-28 6.9l-49.6 60.6A370.66 370.66 0 0 1 130.6 204.11l60.6-49.6a23.94 23.94 0 0 0 6.9-28l-48-112A24.16 24.16 0 0 0 122.6.61l-104 24A24 24 0 0 0 0 48c0 256.5 207.9 464 464 464a24 24 0 0 0 23.4-18.6l24-104a24.29 24.29 0 0 0-14.01-27.6z"
            />
          </svg>
          <span
            class="spacing"
          >
            +91 1234567890
          </span>
        </p>
        <p>
          <svg
            class="contactIcon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
          <span
            class="spacing"
          >
            companyxyz@company.com
          </span>
        </p>
      </div>
    </div>
    <div
      class="footer d-flex "
    >
      <div>
        <img
          alt="carenbit"
          class="carenbit"
          src="carenbit.png"
        />
      </div>
      <p
        class="mediaIcon"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 448c-110.532 0-200-89.451-200-200 0-110.531 89.451-200 200-200 110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200zm107.351-101.064c-9.614 9.712-45.53 41.396-104.065 41.396-82.43 0-140.484-61.425-140.484-141.567 0-79.152 60.275-139.401 139.762-139.401 55.531 0 88.738 26.62 97.593 34.779a11.965 11.965 0 0 1 1.936 15.322l-18.155 28.113c-3.841 5.95-11.966 7.282-17.499 2.921-8.595-6.776-31.814-22.538-61.708-22.538-48.303 0-77.916 35.33-77.916 80.082 0 41.589 26.888 83.692 78.277 83.692 32.657 0 56.843-19.039 65.726-27.225 5.27-4.857 13.596-4.039 17.82 1.738l19.865 27.17a11.947 11.947 0 0 1-1.152 15.518z"
          />
        </svg>
        2024 Carenbit
      </p>
      <div
        class=" d-flex justify-content-between gap-3"
      >
        <span
          class="contactDetails"
        >
          <a
            href="mailto:companyxyz@gmail.com"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              class="mediaIcon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
              />
            </svg>
          </a>
        </span>
        <span
          class="contactDetails"
        >
          <a
            href="https://linkedin.com/in/carenbit"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              class="mediaIcon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
              />
            </svg>
          </a>
        </span>
        <span
          class="contactDetails"
        >
          <a
            href="https://instagram.com/carenbit"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              class="mediaIcon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
              />
            </svg>
          </a>
        </span>
        <span
          class="contactDetails"
        >
          <a
            href="https://twitter.com/carenbit"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              class="mediaIcon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M389.2 48h70.6L305.6 224.2 487 464H345L233.7 318.6 106.5 464H35.8L200.7 275.5 26.8 48H172.4L272.9 180.9 389.2 48zM364.4 421.8h39.1L151.1 88h-42L364.4 421.8z"
              />
            </svg>
          </a>
        </span>
      </div>
    </div>
  </div>
</div>
`;
